moirae/estimators/online/utils/model.py

```diff
@@ -57,18 +80,158 @@ def __init__(self,
 
     @property
     def num_output_dimensions(self) -> int:
         return self._num_output_dimensions
 
 
+class CellModelWrapper(BaseCellWrapper):
+    """
+    Base link between the :class:`~moirae.model.base.CellModel` and the numpy-only interface of
+    :class:`~moirae.estimators.online.filters.base.BaseFilter` filter implementations. This particular wrapper does not
+    touch the A-SOH, but only uses it for predictions.
+    """
+    def __init__(self,
+                 cell_model: CellModel,
+                 asoh: HealthVariable,
+                 transients: GeneralContainer,
+                 inputs: InputQuantities) -> None:
+        super().__init__(cell_model=cell_model, asoh=asoh, transients=transients, inputs=inputs)
+
+        self.num_transients = transients.to_numpy().shape[1]
+
+    @property
+    def num_hidden_dimensions(self) -> int:
+        return self.num_transients
+
+    def update_hidden_states(self,
+                             hidden_states: np.ndarray,
+                             previous_controls: np.ndarray,
+                             new_controls: np.ndarray) -> np.ndarray:
+        """
+        Function that takes a numpy representation of the transient states and updates it based on the cell model, the
+        previous and new controls. Recall the cell model needs the A-SOH as well!
+        """
+        # Convert objects
+        transients = convert_numpy_to_model(filter_array=hidden_states, template=self.transients)
+        previous_inputs = convert_numpy_to_model(filter_array=previous_controls, template=self.inputs)
+        new_inputs = convert_numpy_to_model(filter_array=new_controls, template=self.inputs)
+
+        # Update transients
+        new_transients = self.cell_model.update_transient_state(previous_inputs=previous_inputs,
+                                                                new_inputs=new_inputs,
+                                                                transient_state=transients,
+                                                                asoh=self.asoh)
+
+        return new_transients.to_numpy()
+
+    def predict_measurement(self, hidden_states: np.ndarray, controls: np.ndarray) -> np.ndarray:
+        """
+        Function that takes a numpy representation of the transient state and of the controls, and predicts measurements
+        from it
+        """
+        # Convert objects
+        transients = convert_numpy_to_model(filter_array=hidden_states, template=self.transients)
+        inputs = convert_numpy_to_model(filter_array=controls, template=self.inputs)
+
+        # Get output
+        measurements = self.cell_model.calculate_terminal_voltage(new_inputs=inputs,
+                                                                  transient_state=transients,
+                                                                  asoh=self.asoh)
+
+        return measurements.to_numpy()
+
+
+class DegradationModelWrapper(BaseCellWrapper):

```

Comment:
Could you make tests for these classes as well?

Reply:
On it!